File: problems/0605-can-place-flowers/solution.cpp
// solution in 12 minutes
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        int i = 0;
        
       for (int i = 0; i < flowerbed.size(); i++) {
            if (n <= 0) break;
            if (flowerbed[i] == 1) {
                i++; // skip extra since next one won't be available
                continue;
            }
           
            bool left = i == 0 || flowerbed[i - 1] == 0;
            bool right = i == flowerbed.size() - 1 || flowerbed[i + 1] == 0;
            if (right && left) {
                n--;
                flowerbed[i] = 1;
                i++; // skip one extra index since we know it won't be available
            }
        }
        return n <= 0;
    }
};

File: problems/0605-can-place-flowers/solution.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        count = 0

        for i in range(len(flowerbed)):

            openLeft = i == 0 or flowerbed[i - 1] == 0
            openRight = i == len(flowerbed) - 1 or flowerbed[i + 1] == 0
            
            if openLeft and openRight and flowerbed[i] == 0:
                count += 1
                flowerbed[i] = 1

        return count >= n


